uberduck_ml_dev/models/tacotron2.py

```diff
@@ -713,15 +740,18 @@ def forward(self, inputs):
         )
 
     def inference(self, inputs):
-        text, speaker_ids, *_ = inputs
+        text, input_lengths, speaker_ids, *_ = inputs
 
         embedded_inputs = self.embedding(text).transpose(1, 2)
-        embedded_text = self.encoder.inference(embedded_inputs)
+        embedded_text = self.encoder.inference(embedded_inputs, input_lengths)
 
         encoder_outputs = torch.cat((embedded_text,), dim=2)
 
-        mel_outputs, gate_outputs, alignments = self.decoder.inference(encoder_outputs)
-
+        memory_lengths = input_lengths
+        mel_outputs, gate_outputs, alignments = self.decoder.inference(
+            encoder_outputs, memory_lengths
+        )
+        # mel_outputs, gate_outputs, alignments, mel_lengths = self.decoder.inference(encoder_outputs, memory_lengths)
```

Comment:
I think we need `mel_lengths`. Otherwise users of this function will have to look at `gate_outputs` to figure out how long each batch is, which seems wasteful.

Reply:
where should mel_lengths be called from?